jetty-websocket/websocket-common/src/main/java/org/eclipse/jetty/websocket/common/WebSocketSession.java

```diff
@@ -546,6 +544,9 @@ public void setUpgradeResponse(UpgradeResponse response)
     @Override
     public SuspendToken suspend()
     {
+        if (!isOpen())
```

Comment:
Considering that `connection.suspend()` ultimately just calls `suspending()` in `ReadState`, can't we do all the checks for state in one spot there?  Does calling onClose have an impact on the ReadState?

Reply:
After `onClose` is called the readstate will eventually move from `READING` to `EOF` or `DISCARD`, but that will not happen straight away. So removing this check will cause the `SuspendResumeTest` to fail occasionally because sometimes it allows the suspend after we have been notified of the close as we are still in `READING` state.

The `WebSocketSession` is the only one that knows if the close has been notified so right now we wont know this in `ReadState`.